Phala-Network/index-contract
# Phat contract queries off-chain indexer to check transaction validity

## How to check if a bridge transfer is valid

1. check the source chain side tx by simply checking the nonce
2. check the dest chain side tx by checking
    - if one of the events since the last block info contains the amount that falls in the range specified in the bridge step, return the first one

then return the event's block info, and **set the next bridge step that has the same destination chain as the bridge step just checked**

a vec `LastBlockInfoVec` is used to maintain bridges' block info in their last event

```rust
Vec<(dest_chain_name, LastBlockInfo>
```

when step 2 is finished, the checker returns a `LastBlockInfo`, the `run` function then updates `LastBlockInfoVec`

## Cache restoration

simply reuse the `check` function
